src/utils.ts

```diff
@@ -140,6 +140,46 @@ async function botRedactUserMessagesIn(
     }
 }
 
+/**
+ * Match a user ID or glob against a target ID
+ * @param userId - user ID or glob to test
+ * @param targetId - user ID to test against
+ * @param isGlob - whether the targetID is a glob
+ */
+function matchUserId(userId: string, targetId: string, isGlob: boolean): boolean {
+    const matcher = new MatrixGlob(targetId);
+    if (isGlob) {
```

Comment:
Suggest moving this into the branch, in case it explodes one day:
```suggestion
    if (isGlob) {
        const matcher = new MatrixGlob(targetId);
```